dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -244,18 +245,39 @@ namespace dwa_local_planner {
     base_local_planner::publishPlan(path, g_plan_pub_);
   }
 
-  void DWAPlannerROS::publishScaledFootprint(const geometry_msgs::PoseStamped& pose, const base_local_planner::Trajectory &traj) const {
-    visualization_msgs::Marker marker;
-    marker.header.frame_id = pose.header.frame_id;
-    marker.header.stamp = ros::Time::now();
-    marker.lifetime = ros::Duration(2 * dp_->getSimPeriod()); // double the sim period to avoid flickering
-    marker.type = visualization_msgs::Marker::LINE_STRIP;
-    marker.pose = pose.pose;
-    marker.scale.x = 0.01;
-    marker.color.g = marker.color.a = 1.0; // green
-    marker.points = dp_->getScaledFootprint(traj);
-    marker.points.push_back(marker.points.front()); // close the polygon
-    scaled_fp_pub_.publish(marker);
+  void DWAPlannerROS::publishScaledFootprint(const base_local_planner::Trajectory& traj) const {
+    visualization_msgs::MarkerArray markers;
```

Comment:
F: add premature return to avoid  computation:
`if (scaled_fp_pub_.getNumSubscribers() == 0) return`

Reply:
Fixed